_maps/__MAP_DEFINES.dm

```diff
@@ -40,7 +29,97 @@
+
+#define SPACERUIN_MAP_EDGE_PAD 15
+
+// traits
+// boolean - marks a level as having that property if present
+#define ZTRAIT_CENTCOM "CentCom"
+#define ZTRAIT_MAINSTATION "Main Station"
+#define ZTRAIT_MINING "Mining"
+#define ZTRAIT_RESERVED "Transit/Reserved"
+#define ZTRAIT_AWAY "Away Mission"
+#define ZTRAIT_SPACE_RUINS "Space Ruins"
+#define ZTRAIT_LAVA_RUINS "Lava Ruins"
+#define ZTRAIT_ICE_RUINS "Ice Ruins"
+#define ZTRAIT_ICE_RUINS_UNDERGROUND "Ice Ruins Underground"
+#define ZTRAIT_ISOLATED_RUINS "Isolated Ruins"	//Placing ruins on z levels with this trait will use turf reservation instead of usual placement.
+
+#define ZTRAIT_ASHSTORM "Weather_Ashstorm"	// boolean - weather types that occur on the level
+#define ZTRAIT_GRAVITY "Gravity"	// number - default gravity if there's no gravity generators or area overrides present
+
+// numeric offsets - e.g. {"Down": -1} means that chasms will fall to z - 1 rather than oblivion
+#define ZTRAIT_UP "Up"
+#define ZTRAIT_DOWN "Down"
+
+#define ZTRAIT_LINKAGE "Linkage"	// enum - how space transitions should affect this level
+#define ZTRAIT_BASETURF "Baseturf"	// string - type path of the z-level's baseturf (defaults to space)
+
+// default trait definitions, used by SSmapping
+#define ZTRAITS_CENTCOM list(ZTRAIT_CENTCOM = TRUE)
+#define ZTRAITS_MAINSTATION list(ZTRAIT_LINKAGE = CROSSLINKED, ZTRAIT_MAINSTATION = TRUE)
+#define ZTRAITS_SPACE list(ZTRAIT_LINKAGE = CROSSLINKED, ZTRAIT_SPACE_RUINS = TRUE)
+#define ZTRAITS_LAVALAND list(\
+	ZTRAIT_MINING = TRUE, \
+	ZTRAIT_ASHSTORM = TRUE, \
+	ZTRAIT_LAVA_RUINS = TRUE, \
+	ZTRAIT_BASETURF = /turf/simulated/floor/plating/lava/smooth/lava_land_surface)
+
 // Convenience define
-	#define DECLARE_LEVEL(NAME,LINKS,TRAITS) list("name" = NAME, "linkage" = LINKS, "attributes" = TRAITS)
+#define DL_NAME "name"
+#define DL_TRAITS "attributes"
+#define DL_LINKS "linkage"
+
+#define DECLARE_LEVEL(NAME,LINKS,TRAITS) list(DL_NAME = NAME, DL_LINKS, DL_TRAITS = TRAITS)
+#define AWAY_MISSION_LIST list(\
+	DECLARE_LEVEL(AWAY_MISSION,UNAFFECTED, list(BLOCK_TELEPORT, AWAY_LEVEL)))
+
+// This must match emerald.dm, plus the Away Mission map, plus the Empty Area created in the mapping subsystem, plus the Ruin Area #[i] maps, for things to work correctly
+#define DEFAULT_MAP_TRAITS list(\
+	DECLARE_LEVEL(MAIN_STATION, CROSSLINKED, list(STATION_LEVEL, STATION_CONTACT, REACHABLE, AI_OK)), \
+	DECLARE_LEVEL(CENTCOMM, SELFLOOPING, list(ADMIN_LEVEL, BLOCK_TELEPORT, IMPEDES_MAGIC)), \
+	DECLARE_LEVEL(CONSTRUCTION, CROSSLINKED, list(REACHABLE)), \
+	DECLARE_LEVEL(MINING, SELFLOOPING, list(ORE_LEVEL, REACHABLE, STATION_CONTACT, HAS_WEATHER, AI_OK)), \
+	DECLARE_LEVEL(AWAY_MISSION, UNAFFECTED, list(AWAY_LEVEL, BLOCK_TELEPORT)), \
+	DECLARE_LEVEL("Empty Area", CROSSLINKED, list(REACHABLE)))
+
+#define SPACE_RUINS_NUMBER rand(config.extra_space_ruin_levels_min, config.extra_space_ruin_levels_max)
```

Comment:
This number need to be assigned/reserved ahead of time and I'm not sure if this fits the bill properly.